Content.Server/_Goobstation/StationEvents/GameDirectorSystem.cs

```diff
@@ -0,0 +1,576 @@
+        }
+    }
+
+    /// <summary>
+    ///   Build a list of events to use for the entire story
+    /// </summary>
+    private void SetupEvents(GameDirectorComponent scheduler, PlayerCount count, SelectedGameRulesComponent? selectedRules = null)
+    {
+        scheduler.PossibleEvents.Clear();
+
+        if (selectedRules != null)
+        {
+            SelectFromTable(scheduler, count, selectedRules);
+        }
+        else
+        {
+            SelectFromAllEvents(scheduler, count);
+        }
+        LogMessage($"All possible events added");
+    }
+
+    private void SelectFromAllEvents(GameDirectorComponent scheduler, PlayerCount count)
+    {
+        foreach (var proto in GameTicker.GetAllGameRulePrototypes())
+        {
+            if (!proto.TryGetComponent<StationEventComponent>(out var stationEvent, _factory))
+                continue;
+
+            if (proto.HasComponent<DynamicRulesetComponent>()) // Block john shitcode moment
+                continue;
+
+            // Begin DeltaV - event ignores
+            if (proto.HasComponent<GameDirectorIgnoreComponent>()) // We don't want every event in the game directors
+                continue;
+            // End DeltaV - event ignores
+
+            // Gate here on players, but not on round runtime. The story will probably last long enough for the
+            // event to be ready to run again, we'll check CanRun again before we actually launch the event.
+            if (!_event.CanRun(proto, stationEvent, count.Players, TimeSpan.MaxValue))
+                continue;
+
+            scheduler.PossibleEvents.Add(new PossibleEvent(proto.ID, stationEvent.Chaos));
+        }
+    }
+
+    private void SelectFromTable(GameDirectorComponent scheduler, PlayerCount count, SelectedGameRulesComponent? selectedRules)
+    {
+        if (selectedRules == null)
+            return;
+
+        if(!_event.TryBuildLimitedEvents(selectedRules.ScheduledGameRules, out var possibleEvents))
+            return;
+
+        foreach (var entry in possibleEvents)
+        {
+            var proto = entry.Key;
+            var stationEvent = entry.Value;
+            LogMessage(proto.ID);
+            if (proto.HasComponent<DynamicRulesetComponent>()) // Block john shitcode moment
+                continue;
+            scheduler.PossibleEvents.Add(new PossibleEvent(proto.ID, stationEvent.Chaos));
+        }
+    }
+
+
+    /// <summary>
+    ///   Decide what event to run next
+    /// </summary>
+    protected override void ActiveTick(EntityUid uid, GameDirectorComponent scheduler, GameRuleComponent gameRule, float frameTime)
+    {
+        var currTime = _timing.CurTime;
+        if (currTime < scheduler.TimeNextEvent)
+            return;
+
+        ChaosMetrics chaos = CalculateChaos(uid);
+        scheduler.CurrentChaos = chaos;
+        LogMessage($"Chaos is: {chaos}");
+
+        if (scheduler.Stories == null || scheduler.Stories.Count() <= 0)
```

Comment:
Use the "Length" property instead of Enumerable.Count()
```suggestion
        if (scheduler.Stories == null || scheduler.Stories.Length <= 0)
```